assets/src/edit-story/components/library/panes/media/local/mediaPane.js

```diff
@@ -176,13 +209,27 @@ function MediaPane(props) {
   const onSelect = (mediaPickerEl) => {
     const resource = getResourceFromMediaPicker(mediaPickerEl);
     try {
+      if (
+        enableMediaPickerVideoOptimization &&
+        isFeatureEnabled &&
+        isTranscodingEnabled &&
+        !resource.local &&
+        transcodableMimeTypes.includes(resource.mimeType)
+      ) {
+        optimizeVideo({ resource });
+      }
+
       // WordPress media picker event, sizes.medium.url is the smallest image
       insertMediaElement(
         resource,
         mediaPickerEl.sizes?.medium?.url || mediaPickerEl.url
       );
 
-      if (!resource.posterId && !resource.local) {
+      if (
+        !resource.posterId &&
+        !resource.local &&
+        allowedVideoMimeTypes.includes(resource.mimeType)
```

Comment:
Is a similar check needed also in other places where `uploadVideoPoster()` is being used?

Reply:
I don't believe so. Other places where posters are generated, are after the file is trancoded and uploaded so an MP4 already. it is also work noting, even if the poster generation fails, it does so silently. Poster generation works for some file types if the browser supports them. All that to be said, I don't think we need it other place. Specially now posters are generated and uploaded straight after video upload.